core/src/main/java/com/linecorp/armeria/common/RequestTarget.java

```diff
@@ -131,6 +131,14 @@ static RequestTarget forClient(String reqTarget, @Nullable String prefix) {
      */
     String maybePathWithMatrixVariables();
 
+    /**
+     * Returns the server-side raw path of this {@link RequestTarget} from the ":path" header.
+     * Unlike {@link #path()}, the returned string is the original path without any normalization.
+     * For client-side target it always returns {@code null}.
```

Comment:
Would returning a non-null make sense on the client-side? By doing so, this method could be always non-null, which removes the need for null check on the server side as well (i.e. no complaints from NullAway and other linters)

Reply:
Exposing it only in `ServiceRequestContext` sounds good to me.

Reply:
took a closer look, b/c DefaultRequestTarget and DefaultServiceRequestContext don't live within the same package, it doesn't seems to be very easy, I might be missing something though

Reply:
We could keep `RequestTarget.rawPath()` public and just access it from `DefaultServiceRequestContext`, asserting it is non-null before returning it?

Reply:
that should work! (I was thinking about keeping rawPath package private but java doesn't seem to work as scala), thanks for you idea!

Reply:
updated!